src/couch/src/couch_db_updater.erl

```diff
@@ -1079,14 +1079,13 @@ copy_docs(Db, #db{fd = DestFd} = NewDb, MixedInfos, Retry) ->
                 {Body, AttInfos} = copy_doc_attachments(Db, Sp, DestFd),
                 % In the future, we should figure out how to do this for
                 % upgrade purposes.
-                EJsonBody = case is_binary(Body) of
+                ExternalSize = case is_binary(Body) of
                     true ->
-                        couch_compress:decompress(Body);
+                        couch_compress:uncompressed_size(Body);
                     false ->
-                        Body
+                        ?term_size(Body)
```

Comment:
If you move `?term_size()` into [`couch_compress:uncompressed_size()`](https://github.com/apache/couchdb/pull/835/files#diff-6991b69feab44252db52c58fff0f1812R97), then you can get rid of the `case` and make this `ExternalSize = couch_compress:uncompressed_size(Body),` (at least in my limited testing, you may be covering for cases I haven’t seen)

Reply:
@janl thought of that but noticed how in `decompress` there is an error for anything that's not an external term or snappy compression, so thought of keeping the same structure.